packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,248 @@
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, invocations, abis, details, enableAuthorize } = params;
+
+    const calls = getInvokeCalls(invocations as Invocations);
+
+    const accountDeployed = await isAccountDeployed(this.network, address);
+
+    if (!accountDeployed) {
+      const { callData } = getAccContractAddressAndCallData(
+        this.account.publicKey,
+      );
+
+      const deployAccountpayload = {
+        classHash: ACCOUNT_CLASS_HASH,
+        contractAddress: address,
+        constructorCalldata: callData,
+        addressSalt: this.account.publicKey,
+      };
+
+      invocations.unshift({
+        type: TransactionType.DEPLOY_ACCOUNT,
+        payload: deployAccountpayload,
+      });
+    }
+
+    if (
+      // Get Starknet expected not to show the confirm dialog, therefore,
+      // `enableAuthorize` will set to false to bypass the confirmation
+      // TODO: enableAuthorize should set default to true
+      enableAuthorize &&
+      !(await this.getExecuteTxnConsensus(
+        address,
+        calls,
+        abis,
+        details,
+        accountDeployed,
+      ))
+    ) {
+      throw new UserRejectedRequestError() as unknown as Error;
+    }
+
+    if (!accountDeployed) {
+      await createAccount(
+        this.network,
+        this.account.publicKey,
+        this.account.privateKey,
+        CAIRO_VERSION,
+        true,
+      );
+    }
+
+    const estimateFeeResp = await getEstimatedFees(
+      this.network,
+      address,
+      this.account.privateKey,
+      this.account.publicKey,
+      invocations as unknown as Invocations,
+      details?.version ?? TRANSACTION_VERSION,
+      !accountDeployed,
+    );
+
+    const resp = await executeTxnUtil(
+      this.network,
+      address,
+      this.account.privateKey,
+      calls,
+      abis,
+      {
+        nonce: accountDeployed ? undefined : 1,
+        maxFee: estimateFeeResp.suggestedMaxFee,
+      },
+    );
+
+    if (resp.transaction_hash) {
+      const callData = calls[0].calldata as Calldata;
+
+      const txn: Transaction = {
+        txnHash: resp.transaction_hash,
+        txnType: VoyagerTransactionType.INVOKE,
+        chainId: this.network.chainId,
+        senderAddress: address,
+        contractAddress: calls[0].contractAddress,
+        contractFuncName: calls[0].entrypoint,
+        contractCallData: callData.map((data: string) => {
+          try {
+            return `0x${BigInt(data).toString(16)}`;
+          } catch (error) {
+            // data is already send to chain, hence we should not throw error
+            return '0x0';
+          }
+        }),
```

Comment:
do you know why it will cause error, and does callData is a string already?

Reply:
callData is an array of string. I added enforce in refine of invocationStruct and removed this.

Reply:
nit: it is okay to use the first call as we only support for our dapp